components/resource-creation/ResourceDisplay.tsx

```diff
@@ -84,6 +87,79 @@ function ResourceDisplay() {
     setSelectedDataRequirement({ name: '', content: null });
   };
 
+  const dateForResource = (resource: fhir4.FhirResource ) => {
+    if (!resource || !PrimaryDatePaths?.parsedPrimaryDatePaths) {
+      return 'N/A'
+    }
+
+    const dateinfo =  PrimaryDatePaths.parsedPrimaryDatePaths[resource.resourceType] 
+    if (!dateinfo) {
+      return 'N/A'
+    }
+
+    for (const nameOfResourceDate of Object.keys(dateinfo)) {
+      // If only one dataType
+      const resourceDateData = fhirpath.evaluate(resource, nameOfResourceDate)[0]
+      if(dateinfo[nameOfResourceDate].dataTypes.length == 1 && resourceDateData){
+        // If the only dataType is a period 
+        if(dateinfo[nameOfResourceDate].dataTypes[0] == 'Period'){
+          return formatPeriod(resource as fhir4.Period, nameOfResourceDate, nameOfResourceDate)
+        }
+        // If the only dataType is either dateTime or date
+        else {
+          return (
+            <Tooltip arrowPosition='side' arrowOffset={25} arrowSize={8} label={nameOfResourceDate} withArrow position='top-start'>
+              <Text >{formatDate(fhirpath.evaluate(resource, nameOfResourceDate)[0])}</Text>
+            </Tooltip>  
+          )
+        }
+      }
+      
+      // If isChoiceType is true
+      else {
+        for (const dataType of dateinfo[nameOfResourceDate].dataTypes) {
+          // Capitalize the first char of dataType and append for proper resource date name
+          const fullResourceDateName = nameOfResourceDate + dataType.charAt(0).toUpperCase() + dataType.slice(1);
+          if (fhirpath.evaluate(resource, fullResourceDateName)[0]) {
+            if (dataType == 'Period') {
+              return formatPeriod(resource as fhir4.Period, fullResourceDateName, fullResourceDateName);
+            }
+            // Else if dataType == 'dateTime' || 'Date'
+            else {
+              return (
+                <Tooltip arrowPosition='side' arrowOffset={25} arrowSize={8} label={fullResourceDateName} withArrow position='top-start'>
+                  <Text>{formatDate(fhirpath.evaluate(resource, fullResourceDateName)[0])}</Text>
+                </Tooltip>
+              )
+            }
+          }
+        }
+      }
+    }
+    return 'N/A'
+  }
+
+
+  // Formatter for if the date is a period
+  const formatPeriod = (resource: fhir4.Period, resourcePeriod: string, dateTypeName: string) => {
```

Comment:
Also it looks like there is some mismatch with variable names and types here...I think that the resource name of the fhir4.Period is potentially misleading, I would change that to period: fhir4.Period or something.

Reply:
On further investigation, I think resource should be of type fhir4.FhirResource